### ISSUE TITLE
wxComboCtrl does not support dark mode on MSW

### ISSUE THREAD

**Issue by PBfordev** [wxWidgets/wxWidgets#23766]
### Description
`wxComboCtrl`/`wxOwnerDrawnComboBox` on Windows using dark mode do not use the dark colors properly. The issue can be seen in the unmodified combo sample:
![wx-combo-darkmode](https://github.com/wxWidgets/wxWidgets/assets/12495521/4923cae3-14e9-4246-a795-b42b7d6581a9)

If the issue cannot be fixed, perhaps it should be added to the list of known issues in MSW dark mode in `wxApp::MSWEnableDarkMode()` docs.

To Reproduce:
1. Make sure Windows is set to dark mode and environment variable `wx_msw_dark_mode` is set to `1`.
2. Run the combo box sample.
3. From the menu choose "Compare against wxComboBox...".
4. Observe the differences between the controls on the left and right side of the dialog.

### Platform and version information
- wxWidgets version: current GIT master
- wxWidgets port: wxMSW
- OS: Windows 10 22H2 v19045.3324

**Comment by rudolfwalter** [comment]
Something like the following gets us halfway there:

```patch
diff --git a/src/msw/combo.cpp b/src/msw/combo.cpp
index 21f3ad3f16..009ab8dc25 100644
--- a/src/msw/combo.cpp
+++ b/src/msw/combo.cpp
@@ -35,6 +35,7 @@
 #include "wx/msw/registry.h"
 #include "wx/msw/uxtheme.h"
 #include "wx/msw/dc.h"
+#include "wx/msw/private/darkmode.h"

 #define NATIVE_TEXT_INDENT_XP       4
 #define NATIVE_TEXT_INDENT_CLASSIC  2
@@ -110,6 +111,8 @@ bool wxComboCtrl::Create(wxWindow *parent,
     if ( style & wxCC_STD_BUTTON )
         m_iFlags |= wxCC_POPUP_ON_MOUSE_UP;

+    wxMSWDarkMode::AllowForWindow(m_hWnd, L"CFD");
+
     // Create textctrl, if necessary
     CreateTextCtrl( wxNO_BORDER );

```

Before:
<img width="700" alt="Image" src="https://github.com/user-attachments/assets/3301d627-57f6-4dc9-baa6-92f0b81efd88" />

After:
<img width="700" alt="Image" src="https://github.com/user-attachments/assets/e3364b59-6622-4414-a879-8e033726828f" />
  
This is still not perfect, because the text field and actual combo have different shades of grey.

Also, it seems that this causes any wxOwnerDrawnComboBox which calls SetButtonBitmaps(...,true,...) to now be drawn with a light dropdown arrow (whereas without this that single piece is drawn dark!), as you can see in the bottom-right combo box in the screenshot.

PS: This is what the "Compare" dialog looks like with this:
![Image](https://github.com/user-attachments/assets/71e3736d-fcd5-4990-8fad-420d9011eba9)

**Comment by vadz** [comment]
Yes, using "CFD" theme makes sense (even though I still have no idea what does it stand for — but we already use it for `wxChoice`). The rest of the issues can probably be dealt with at wx level...

**Comment by rudolfwalter** [comment]
The only thing which abbreviates to "CFD" that I'm aware of are the _Common File Dialogs_.  ¯\\\_(ツ)_/¯

**Comment by vadz** [comment]
> The only thing which abbreviates to "CFD" that I'm aware of are the _Common File Dialogs_. 

Wow, this must be it, thanks, I had never thought about it.

I've tried looking at the background problem, but without success so far: `wxComboCtrl::PrepareBackground()` seems to be completely irrelevant, background is completely overwritten in `wxComboCtrl::OnPaintEvent()` anyhow. And the latter uses `::DrawThemeBackground()` to draw it, which draws this wrong grey, presumably because the code uses "COMBOBOX" theme. Replacing it with "CFD" does result in correct background — but without any borders. I think we could overwrite the background inside the borders, but it's still weird that "CFD" doesn't draw them at all. Anyhow, I'll try to return to this later.

**Comment by vadz** [comment]
> > The only thing which abbreviates to "CFD" that I'm aware of are the _Common File Dialogs_.
> 
> Wow, this must be it, thanks, I had never thought about it.

And this has actual consequences: because it's this (and not "Combobox For Dark mode" for example), it's not specific to comboboxes and so we must specify "COMBOBOX" as the class to use explicitly to get the correct behaviour from theme API for combobox-specific parts such as `CB_BORDER`. Adding it makes the borders use the same colour as in the native control in dark mode.

> I've tried looking at the background problem

Unfortunately I'm still not sure what to do about this or even if it's a problem at all because if you look at the comboboxes in MSVS in dark mode, they have the same `#383838` background as ours do, with the change here — even though text entries in MSVS use `#1f1f1f`, which is very close to `#202020` used by `wxTextCtrl` in dark mode. The only way to use the same `#202020` background for the combobox seems to be to draw it ourselves. The alternative would be to set the text control background to the one used by the combobox. But we still need to draw the background ourselves if the background is explicitly changed with `SetBackgroundColour()`, so it's simpler to just do it in all cases and this is what I ended up doing in #25607, please test it.

Here is the result on my test system, AFAICS it's good enough, but please let me know if you see anything seriously wrong:

<img width="1006" height="1148" alt="Image" src="https://github.com/user-attachments/assets/64857f93-ffd4-4043-9980-767c230a5712" />
